worker/src/test/java/org/verapdf/webapp/worker/listener/ExecutableTaskWorkerTests.java

```diff
@@ -0,0 +1,555 @@
+package org.verapdf.webapp.worker.listener;
+
+import org.apache.commons.io.FileUtils;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.ArgumentMatcher;
+import org.mockito.Mockito;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.core.io.FileSystemResource;
+import org.springframework.core.io.Resource;
+import org.springframework.http.HttpStatus;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.web.client.RestClientResponseException;
+import org.verapdf.webapp.localstorageservice.client.service.LocalStorageServiceClient;
+import org.verapdf.webapp.localstorageservice.model.dto.StoredFileDTO;
+import org.verapdf.webapp.queueclient.listener.QueueListener;
+import org.verapdf.webapp.queueclient.sender.QueueSender;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.ArgumentMatchers.*;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.util.Arrays;
+import java.util.UUID;
+
+@ActiveProfiles("test")
+@SpringBootTest
+public class ExecutableTaskWorkerTests {
+
+	@Autowired
+	private ExecutableTaskWorker executableTaskWorker;
+
+	@Autowired
+	private File workingDir;
+
+	@MockBean
+	private LocalStorageServiceClient localStorageServiceClient;
+
+	@MockBean
+	private QueueSender queueSender;
+
+	@MockBean
+	private QueueListener queueListener;
+
+	@AfterEach
+	public void checkWorkingDirectory() {
+		assertTrue(workingDir.listFiles().length == 0);
+		Mockito.verifyNoMoreInteractions(localStorageServiceClient, queueSender);
+		Mockito.validateMockitoUsage();
+	}
+
+	@Test
+	public void processTest() throws URISyntaxException, IOException {
+		UUID fileId = UUID.fromString("07b17d5d-1010-4980-b280-94129cb13838");
+
+		Mockito.doReturn(createDescriptor(fileId))
+				.when(localStorageServiceClient)
+				.getFileDescriptorById(fileId);
+		Mockito.doReturn(createResource())
+				.when(localStorageServiceClient)
+				.getFileResourceById(fileId);
+		byte[] correctReportBytes = getClass().getResourceAsStream(
+				"/files/report-PDFA_1_A.json").readAllBytes();
+		ArgumentMatcher<File> correctResource = file -> {
+			try {
+				return Arrays.equals(correctReportBytes, FileUtils.readFileToByteArray(file));
+			} catch (IOException e) {
+				throw new IllegalStateException(e);
+			}
+		};
+		Mockito.doReturn(createReportFileDescriptor(fileId))
+				.when(localStorageServiceClient)
+				.saveFile(argThat(correctResource), eq("35a7d1deffc1ad490bcc5eb3c557a399"));
```

Comment:
argThat call should be made on verify (as we are verifying usage there) instead of here

Reply:
the same valid for all further similar cases

Reply:
Can't add it to verify, because when method handleMessage() will finish this file will be dropped and i can't check what this file contains with a help of verify() during test execution